code/components/voip-mumble/src/MumbleClient.cpp

```diff
@@ -980,6 +962,27 @@ MumbleConnectionInfo* MumbleClient::GetConnectionInfo()
 
 void MumbleClient::HandlePing(const MumbleProto::Ping& ping)
 {
+	m_inFlightTcpPings = 0;
+	if (m_crypto.IsInitialized())
+	{
+		// Mimic mumbles behavior for pings
+		m_crypto.m_remoteGood = ping.good();
+		m_crypto.m_remoteLate = ping.late();
+		m_crypto.m_remoteLost = ping.lost();
+		m_crypto.m_remoteResync = ping.resync();
+
+		if (m_hasUdp && (m_crypto.m_remoteGood == 0 || m_crypto.m_localGood == 0) && (msec() - m_lastUdp) > 2s)
+		{
+			console::PrintWarning("mumble", "UDP packets can *not* be received. Switching to TCP tunnel mode.\n");
+			m_hasUdp = false;
+		}
+		else if (!m_hasUdp && m_crypto.m_remoteGood > 3 && m_crypto.m_localGood > 3)

```

Comment:
maybe there should be an additional check for initialization, from what i understand this means that on start we will have to wait for 3 correct ping before it can switch to udp packet ? or maybe i'm missing something ?

Reply:
That is what this means, this shouldn't really be much of an issue considering on the initial setup you'll still be loading in, and a ping is sent every second by the server, which means it would be 3 seconds maximum where the client can't use UDP (this copies the behavior of mumble-server)